#!/usr/bin/env python

from dynamic_reconfigure.parameter_generator_catkin import *

PACKAGE_NAME = "squirrel_navigation"

gen = ParameterGenerator()

# TrajectoryPlanner parameters
gen.add("xy_smoother", double_t, 0, "Trajectory linear smoother", 0.5, 0.0, 1.0)
gen.add("yaw_smoother", double_t, 0, "Trajectory angular smoother", 0.1, 0.0, 1.0)
gen.add("heading_lookahead", int_t, 0, "Lookahead for smoothing the path", 15, 0, 100)
gen.add("time_scaler", double_t, 0, "Time scaling for trajectory generation", 0.5, 0.0, 10.0)

exit(gen.generate(PACKAGE_NAME, "squirrel_navigation", "TrajectoryPlanner"))